File: fromchatgpt/claude_code_fhr_project_bundle/code/python/00_setup_paths.py
"""
Setup and path configuration for SSJDA 1331 analysis.
All data paths are strictly read-only. All outputs go to derived/ directories.

Usage:
  from setup_paths import SSJDA_CSV_FILE, DATA_DERIVED, ...
  or: exec(open("setup_paths.py").read())
"""

import os
from pathlib import Path

# Project root (this is the bundle directory)
PROJECT_ROOT = Path(__file__).parent.parent.parent

# Read-only authoritative source-data directory
AUTHORITATIVE_DATA_DIR = Path(r"C:\Users\yuyaa\git\postwar_household_insurance\postwar_household_insurance\data\raw")

# Specific data files within the source directory
SSJDA_EXTRACT_DIR = AUTHORITATIVE_DATA_DIR / "神奈川県における民生基礎調査（ボーダー・ライン層調査）1961" / "1331"
SSJDA_DTA_FILE = SSJDA_EXTRACT_DIR / "1331.dta"
SSJDA_CSV_FILE = SSJDA_EXTRACT_DIR / "1331.csv"
SSJDA_LABELS_FILE = SSJDA_EXTRACT_DIR / "1331label.txt"
SSJDA_README_FILE = SSJDA_EXTRACT_DIR / "1331readme.docx"

# Project output directories
DATA_DERIVED = PROJECT_ROOT / "data" / "derived"
OUTPUT_TABLES = PROJECT_ROOT / "output" / "tables"
OUTPUT_FIGURES = PROJECT_ROOT / "output" / "figures"
OUTPUT_LOGS = PROJECT_ROOT / "output" / "logs"
DOCS = PROJECT_ROOT / "docs"

# Create derived directories if they don't exist
for d in [DATA_DERIVED, OUTPUT_TABLES, OUTPUT_FIGURES, OUTPUT_LOGS, DOCS]:
    d.mkdir(parents=True, exist_ok=True)

if __name__ == "__main__":
    print("=" * 70)
    print("PATH CONFIGURATION")
    print("=" * 70)
    print(f"Project root: {PROJECT_ROOT}")
    print(f"Read-only data dir: {AUTHORITATIVE_DATA_DIR}")
    print(f"SSJDA CSV file exists: {SSJDA_CSV_FILE.exists()}")
    print(f"SSJDA labels exists: {SSJDA_LABELS_FILE.exists()}")
    print(f"\nOutput directories created.")

    # Verify that source files exist
    if not SSJDA_CSV_FILE.exists():
        raise FileNotFoundError(f"SSJDA CSV file not found: {SSJDA_CSV_FILE}")
    if not SSJDA_LABELS_FILE.exists():
        raise FileNotFoundError(f"SSJDA labels file not found: {SSJDA_LABELS_FILE}")

    print("All paths configured and verified.")

File: fromchatgpt/claude_code_fhr_project_bundle/code/00_setup_paths.R
# Setup and path configuration for SSJDA 1331 analysis
# All data paths are strictly read-only. All outputs go to derived/ directories.

# ============================================================================
# PATHS
# ============================================================================

# Project root
project_root <- here::here()

# Authoritative read-only data directory
authoritative_data_dir <- "C:/Users/yuyaa/git/postwar_household_insurance/postwar_household_insurance/data/raw"

# SSJDA 1331 extract directory
ssjda_extract_dir <- file.path(
  authoritative_data_dir,
  "神奈川県における民生基礎調査（ボーダー・ライン層調査）1961",
  "1331"
)

# Specific data files
ssjda_csv_file <- file.path(ssjda_extract_dir, "1331.csv")
ssjda_dta_file <- file.path(ssjda_extract_dir, "1331.dta")
ssjda_labels_file <- file.path(ssjda_extract_dir, "1331label.txt")
ssjda_readme_file <- file.path(ssjda_extract_dir, "1331readme.docx")

# Project output directories
data_derived <- file.path(project_root, "data", "derived")
output_tables <- file.path(project_root, "output", "tables")
output_figures <- file.path(project_root, "output", "figures")
output_logs <- file.path(project_root, "output", "logs")
docs <- file.path(project_root, "docs")

# Create directories if they don't exist
dir.create(data_derived, recursive = TRUE, showWarnings = FALSE)
dir.create(output_tables, recursive = TRUE, showWarnings = FALSE)
dir.create(output_figures, recursive = TRUE, showWarnings = FALSE)
dir.create(output_logs, recursive = TRUE, showWarnings = FALSE)
dir.create(docs, recursive = TRUE, showWarnings = FALSE)

# Verify source files exist
if (!file.exists(ssjda_csv_file)) {
  stop(paste("SSJDA CSV file not found:", ssjda_csv_file))
}

cat("================================================================================\n")
cat("PATH CONFIGURATION\n")
cat("================================================================================\n")
cat("Project root:", project_root, "\n")
cat("Read-only data dir:", authoritative_data_dir, "\n")
cat("SSJDA CSV file:", ssjda_csv_file, "\n")
cat("\nOutput directories:\n")
cat("  Derived data:", data_derived, "\n")
cat("  Tables:", output_tables, "\n")
cat("  Figures:", output_figures, "\n")
cat("  Logs:", output_logs, "\n")
cat("  Docs:", docs, "\n")
cat("\n[OK] All paths configured and verified.\n")
